src/coreclr/src/vm/class.cpp

```diff
@@ -1067,10 +1090,7 @@ void ClassLoader::ValidateMethodsWithCovariantReturnTypes(MethodTable* pMT)
         {
             GCX_COOP();
```

Comment:
It might be better to move GCX_COOP inside the IsCompatibleWith helper when needed (not all checks need this)

Reply:
Thanks Fadi, it makes sense. I've made that change. Btw, I am still using your branch for finishing this work, so I'd like to ask you to keep it until it is merged.

Reply:
Thanks Jan, yes I’ll keep the branch around